test/reduce-e2e/testdata/join-reduce-pipeline.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: join-on-reduce
+spec:
+  limits:
+    readBatchSize: 50
```

Comment:
Any reason we cannot have the default `readBatchSize`?

Reply:
no, I must have been imitating another test which set it

Reply:
I see the other tests in this directory have it set - should they set it and not this one?